R-package/R/xgb.train.R

```diff
@@ -371,27 +399,32 @@ xgb.train <- function(params = list(), data, nrounds, watchlist = list(),
   # The tree updating process would need slightly different handling
   is_update <- NVL(params[['process_type']], '.') == 'update'
 
+  past_evaluation_log <- NULL
+  if (inherits(xgb_model, "xgb.Booster")) {
+    past_evaluation_log <- attributes(xgb_model)$evaluation_log
+  }
+
   # Construct a booster (either a new one or load from xgb_model)
-  handle <- xgb.Booster.handle(
+  bst <- xgb.Booster(
     params = params,
     cachelist = append(watchlist, dtrain),
     modelfile = xgb_model,
-    handle = NULL
+    training_continuation = training_continuation
+  )
+  niter_init <- bst$niter
+  bst <- bst$bst
+  .Call(
+    XGBoosterCopyInfoFromDMatrix,
+    xgb.get.handle(bst),
+    dtrain
   )
-  bst <- xgb.handleToBooster(handle = handle, raw = NULL)
 
   # extract parameters that can affect the relationship b/w #trees and #iterations
-  num_class <- max(as.numeric(NVL(params[['num_class']], 1)), 1)
-  num_parallel_tree <- max(as.numeric(NVL(params[['num_parallel_tree']], 1)), 1)
+  # Note: it might look like these aren't used, but they need to be defined in this
+  # environment for the callbacks for work correctly.
+  num_class <- max(as.numeric(NVL(params[['num_class']], 1)), 1) # nolint
+  num_parallel_tree <- max(as.numeric(NVL(params[['num_parallel_tree']], 1)), 1) # nolint
```

Comment:
are these still useful after the `ntree_limit` is removed?

Reply:
No, but they will be removed in the follow-up PR.

Reply:
Got it, thank you for sharing.